test/unified_malloc_framework/memoryPool.hpp

```diff
@@ -84,6 +115,209 @@ TEST_P(umfPoolTest, pow2AlignedAlloc) {
     }
 }
 
+TEST_P(umfPoolTest, freeNullptr) {
+    void *ptr = nullptr;
+    auto ret = umfPoolFree(pool.get(), ptr);
+    ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+}
+
+TEST_P(umfPoolTest, allocOutOfMem) {
+    static constexpr size_t numProviders = 1;
+    size_t numProvidersRet = 0;
+    std::array<umf_memory_provider_handle_t, numProviders> retProviders;
+
+    auto ret = umfPoolGetMemoryProviders(pool.get(), numProviders,
+                                         retProviders.data(), &numProvidersRet);
+    ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+    ASSERT_EQ(numProvidersRet, numProviders);
+
+    for (auto provider : retProviders) {
+        if (std::string(umfMemoryProviderGetName(provider)) ==
+            std::string("null")) {
+            GTEST_SKIP();
+        }
+    }
+
+    // test whether memory is kept in a pool accordingly to MaxSize
+    static constexpr size_t allocSize = 16;
```

Comment:
how did you come up with those values? aren't those specific to some memory pool?